intake/source/csv.py

```diff
@@ -41,12 +41,23 @@ def __init__(self, urlpath, csv_kwargs=None, metadata=None,
         super(CSVSource, self).__init__(metadata=metadata,
                                         cache=cache)
 
+    def _get_cache(self):
```

Comment:
Is there any way that this can happen in DataSource base, so that it doesn't need to be reimplemented for every subclass?

Reply:
Well, the implementation could be different for each driver.  Thinking of the case we discussed for parquet with multiple cache specs.

Reply:
I think they will tend to look remarkably similar/identical for almost all of the plugins. Those with special requirements would have to override *or* we have a more general way to specify paths to download. We had some more general ideas in conversation before.

Reply:
Sure, we can certainly have a _default_ implementation.